plugins/module_utils/network/common/utils.py

```diff
@@ -771,3 +772,18 @@ def convert_doc_to_ansible_module_kwargs(doc):
         if item in VALID_ANSIBLEMODULE_ARGS:
             spec.update({item: doc_obj[item]})
     return spec
+
+
+def remove_unwanted_characters(value):
+    printable = set(string.printable)
+    if isinstance(value, str):
+        value = "".join(filter(lambda x: x in printable, value))
+        value = re.sub(r"\[m", "", value)
+        return value
+    elif isinstance(value, list):
+        val = []
+        for el in value:
+            v = "".join(filter(lambda x: x in printable, el))
+            v = re.sub(r"\[m", "", v)
+            val.append(v)
+        return val
```

Comment:
Doesn't this miss a default case where it just returns value ?
For when it's not a string or a list

```suggestion
        return val
      else:
        return value        
```